### PR TITLE
Add custom authorization header for OAuth2.0 strategy

### PR DESCRIPTION
**Pull request by sonali-mishra-94**
Signed-off-by: Sonali Mishra <sonali.priyam@gmail.com>

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: .changeset/thin-poems-watch.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+---
+'@backstage/plugin-auth-backend': patch
+---
+
+---
+
+## '@backstage/plugin-auth-backend': patch
```
Comment (Rugvip) [inline .changeset/thin-poems-watch.md:7]: This needs a bit of cleanup

**Review by Rugvip — COMMENTED** [review]
Thank you! 👍 

We'd need DCO signed, there are some instructions to help with that if you click through to details on the failing check

File: .changeset/thin-poems-watch.md (hunk 1 above)
Comment (Rugvip) [reply .changeset/thin-poems-watch.md:1]: This isn't quite formatted properly still. Could look at one of the other changesets or create a new one using the CLI

Comment (freben) [reply .changeset/thin-poems-watch.md:1]: expected format is

```
---
'@backstage/plugin-auth-backend': patch
---
```

Comment (Rugvip) [reply .changeset/thin-poems-watch.md]: Pushed a fix so we can get this in today's release

**Review by Rugvip — APPROVED** [review]

**Pull request merged**